pkg/cmd/cli/cmd/newapp.go

```diff
@@ -271,15 +274,56 @@ func setLabels(labels map[string]string, result *newcmd.AppResult) error {
 	return nil
 }
 
+// isInvalidTriggerError returns true if the given error is
+// a validation error that contains 'trigger' in its error message.
+// This error is returned from an older server when
+// it doesn't recognize a newer trigger type.
+func isInvalidTriggerError(err error) bool {
+	if !kapierrors.IsInvalid(err) {
+		return false
+	}
+	statusErr, ok := err.(*kapierrors.StatusError)
+	if !ok {
+		return false
+	}
+	return strings.Contains(statusErr.Status().Message, "trigger")
+}
+
+// retryBuildConfig determines if the given error is caused by an invalid trigger
+// error on a BuildConfig. If that is the case, it will remove all triggers with a
+// type that is not in the whitelist for an older server.
+func retryBuildConfig(info *resource.Info, err error) runtime.Object {
+	triggerTypeWhiteList := map[buildapi.BuildTriggerType]struct{}{
+		buildapi.GitHubWebHookBuildTriggerType:  struct{}{},
+		buildapi.GenericWebHookBuildTriggerType: struct{}{},
+		buildapi.ImageChangeBuildTriggerType:    struct{}{},
```

Comment:
> oc new-app throws an error saying she's got an invalid trigger (because the client ignored the fooTrigger part it doesn't know about)

It wouldn't happen with `new-app`, since the objects are all new, but it might happen with an old client updating an existing object against a new server, since the client drops fields it doesn't know about in round-tripping... so FooParams would get deleted and type="Foo" would remain in the content sent to the server... should a new server prevent the update, accept the nil FooParams field, or drop the trigger entirely?